### ISSUE TITLE
Web-RDP canvas size not using correct width on Chrome

### ISSUE THREAD

**Issue by energywave** [Ylianst/MeshCentral#4701]
**Describe the bug**
When connecting Web-RDP using Chrome with the default option "Canvas size" the size of the RDP session use canvas height but not width.
(hope that "Canvas size" it's the english correct definition of "dimensioni della tela" in Italian, it's the first, default option)

**To Reproduce**
1. Open Meshcentral using Google Chrome
2. In Desktop tab click on Settings button
3. On the first field select "canvas size"
4. Click OK
5. Click on RDP connection

**Expected behavior**
I expect, as it did before (cannot remember on what version), that the full extent of the available size in the browser would be used to set the desktop resolution of the RDP connection, some versions ago it was perfectly working.
That was in my opinion the best choice to use all the space by preserving pixel perfect image without any scaling. I really miss that feature.

**Screenshots**
![image](https://user-images.githubusercontent.com/2051567/199474285-db96e4e9-345a-44cb-a9db-5f3a09ee9fc0.png)
As you see the size of the available size is not used in width. It's something like it use full height and then calculate the width to obtain an anachronistic 4:3 aspect rate (just a guess).

**Server Software (please complete the following information):**
 - OS: Windows Server 2012 standard
 - Virtualization: Hyper-V server 2012
 - Network: DMZ with NAT on firewall
 - Version: 1.0.95
 - Node: 16.15.1

**Client Device (please complete the following information):**
 - Device: Desktop
 - OS: Windows 10 Pro 22H2
 - Network: local with meshcentral (not relevant)
 - Browser: Google Chrome 106.0.5249.121
 - MeshCentralRouter Version: not inherent

**Remote Device (please complete the following information):**
 - Device: Servers
 - OS: Windows 2016 datacenter edition and other versions randomly picked
 - Network: remote over WAN
 - Current Core Version (if known): Aug 25 2022, 461107747

**Additional context**
I've tested the same using Firefox ver.106.0.3 64bit and it works correctly. So it's something related to Google Chrome only. 

**Your config.json file**
```
{
	"settings": {
		"Cert": "*******.***",
		"Port": 444,
		"RedirPort": 81,
		"_minify": true
	},
	"smtp": {
		"host": "***.***.***.***",
		"port": 25,
		"from": "*****@********.***",
		"tls": false,
		"tlscertcheck": false,
		"tlsstrict": false
	},
	"domains": {
		"": {
			"assistantCustomization": {
				"title": "*****************",
				"image": "**********",
				"filename": "************"
			}
		}
	}
}
```

**Comment by Ylianst** [comment]
If you go into the "Settings" of the desktop tab and click on RDP, you should be able to set the size of the RDP display.

![image](https://user-images.githubusercontent.com/1319013/199557850-a4f7eed6-903b-420b-bcae-11cc6b869dec.png)

If you select "Canvas Size" then RDP should size to the current window size. Let me know if this works.

**Comment by energywave** [comment]
> If you select "Canvas Size" then RDP should size to the current window size. Let me know if this works.
Yes, I know, and it works. But in the issue I'm alerting you that "Canvas Size" does not work correctly on Chrome. The width is not the one of the canvas size.
If I manually set a resolution from the list, that works correctly, I'll never have a pixel perfect graphical surface but there will always be an interpolation that keep the image less clear and nice.
Canvas Size was so convenient and easy to use, it's a pity that it don't works correctly anymore.

**Comment by Ylianst** [comment]
Oh, I see. Ok, let me check.

**Comment by Ylianst** [comment]
Arg. This is not good, I can't make this problem happen. I have latest Chrome Version 107.0.5304.88 (Official Build) (64-bit) and it always connects RDP with the correct canvas size to fill the screen. I need some way to replicate this issue.

**Comment by energywave** [comment]
Ok, thank you for your trial! I'll test again on my side to identify the problem, without extensions and from different clients. Hope that's not something on my side that wasted your time... :(

**Comment by energywave** [comment]
Ok, found the origin of the problem. Tried Chrome without extensions, same bad behavior. Tried in anonymous mode: correct canvas size!
So I tried to remove cookies and local archives from Chrome for my meshcentral site, logged in again and puff: it works correctly.
So there was something in cookies or local archives that forced that bad behavior but I cannot help identify it anymore as I cleared cookies and local archive :'(
If you want me to do some test in case you're interested in hunting this, I'm available.
Thank you very much for your time and sorry for having wasted your time. Keep on with your wonderful work!!! :)

**Comment by Ylianst** [comment]
Oh, interesting. Let's keep it open for a bit in case other's see this issue.

**Comment by energywave** [comment]
I'm having this issue again! That's good as I can do some debug for you to understand what's happening. If you have suggestions on what to check I can do it :)

**Comment by energywave** [comment]
@Ylianst I've updated to 1.0.98 but still the issue persist. As I'm having the issue again, if you tell me what to check I can send here log or whatever you'll tell me to check so that you fix this problem.

**Comment by silversword411** [comment]
How many cookies are set in chrome? Delete one at a time till is starts working again.

**Comment by energywave** [comment]
Ok, I've tested a bit.
This is the cookies status on Chrome:
![image](https://user-images.githubusercontent.com/2051567/206728526-cc134ef5-54d5-4edb-8727-ad1c6ad0ce23.png)
If I remove the "local archive" ("Archiviazione locale" in Italian) entry and refresh the page then the RDP connection is again full screen.
I really don't know what Local Archive means...
If it behaves like in the past it will be wrong in some days and then I can make other tests, if someone has some idea on what to test.

**Comment by energywave** [comment]
Ok, I narrowed down the issue and I think I've got it!

The issue arrives when I go to RDP settings and I set the option to reverse mouse wheel (I need to do it on every session, otherwise the wheel is wrong direction...)
When I do so the next session will have the canvas size problem.
So I've inspected the local archive and found that:
![Local archive - tela](https://user-images.githubusercontent.com/2051567/206756939-13195161-9ffd-4ba6-b71f-0b4df0aa0e06.PNG)
As you see the `rdpsize `is set to `"tela"` that's so strange because that's an Italian word. Another value appears: `scaling: "1024"`
So I've checked the settings panel and found that the select html object is as follow:
![Display size](https://user-images.githubusercontent.com/2051567/206757088-c70a96fa-0aa4-4559-921a-a505b51d27cc.PNG)
As you see we have values like `"tela"` and `"schermo"`, actual italian words.

**I suspect then that the problem is in translating values of that select...**

I've tried to remove the rdpsize property from desktopsettings value and then, after a refresh of the page, the session is again correct.
But when I go to RDP settings to set the reverse moouse wheel (or any other properties) and I confirm I'm back again.

**Comment by danielgaou** [comment]
The same issue occurs with the Spanish language. `rdpsize` shows `"lienzo"` instead of `"canvas"`. Manually changing this value and refreshing the page makes it display as expected.

**Comment by energywave** [comment]
I"ve tried to fix that but could not find where is the code that use the translated content for canvas... 😓

**Comment by si458** [comment]
I will have a look when I can during the next week for urs

**Comment by si458** [comment]
ok so whats happening is people has added the translation word for `canvas` in `translate.json` so its being replaced
but whats also more annoying is the was a mix of values,
for example "1024x768" was also having mixed translations as 1024x768 (but the x is like a special character)
so ive fixed the translate.json and the handlebars file here https://github.com/Ylianst/MeshCentral/commit/1d67172dd309c151ed941f08324b3a501da40e00
will be in docker `master` repo in about 15mins to test and be in next release!

**Comment by energywave** [comment]
I've tested the new release that should have fixed the issue. Unfortunately it isn't (sorry I could not install a test server to test it before the release...)
I made a test by deleting the "desktopsettings" from "local archive" of developer tools, connected with web-rdp, changed some settings (reversing mouse wheel), disconnected, reconnected. Desktop switched to 1024x768 as the property "rdpsize" is again "tela".
Here's a dump of my current (after that procedure) desktopsettings property:
```
{"encoding":"2",
"showfocus":false,
"showmouse":true,
"showcad":true,
"quality":"40",
"scaling":"1024",
"framerate":"50",
"localkeymap":false,
"swapmouse":false,
"remotekeymap":false,
"autoclipboard":false,
"autolock":false,
"agentencoding":"4",
"rdpflags":47,
"rmw":false,
"kvmrmw":false,
"rdpsize":"tela",
"rdpsmb":false,
"rdprmw":true,
"rdpautoclipboard":false}
```

As you see the "rdpsize" is "tela", that's the translation of "canvas" in italian language.

**Comment by si458** [comment]
@energywave what language are you using sorry?

**Comment by si458** [comment]
FFS its put it back into none english!
<img width="447" alt="image" src="https://github.com/user-attachments/assets/46a5fc4f-2f86-4d66-a7f1-fc8691856b2f">
im going to have to fix the bloody translate tool now!

**Comment by si458** [comment]
how did you install? 
did you `npm install` or docker image?
when i run the translate manually its working and creating the files correctly!
so i think the must be a problem with the npm repo!

**Comment by si458** [comment]
ok its the `npm` repo thats got the issue in, @Ylianst is going to sort it!
might need a new release, might be able to repush a npm release, wait and see!

**Comment by energywave** [comment]
> @energywave what language are you using sorry?

Italian

> how did you install? did you `npm install` or docker image? when i run the translate manually its working and creating the files correctly! so i think the must be a problem with the npm repo!

Uhm, I don't know really... I've used the provided upgrade through the web UI, I don't know what it uses under the hood (but you know it for sure ;) )

> ok its the `npm` repo thats got the issue in, @Ylianst is going to sort it! might need a new release, might be able to repush a npm release, wait and see!

Nice that you sorted out! When an update will be available I'll check it and update here the results.

**Comment by si458** [comment]
all fixed! 1.1.32 👍
